### PR TITLE
touch up "travel times" screen (formerly "trip summaries")

### PR DESCRIPTION
**Pull request by michaelkirk**
## bar chart

Primarily, use more human bucket intervals, like with the "risk exposure" viz. This also puts the long tail of trips into a single bucket. 

**before**
Screen Shot 2021-05-13 at 6 14 00 PM

**after**
Screen Shot 2021-05-13 at 6 05 22 PM

## scatter plot

ui polish: remove undesirable left padding, tweak title

**before**
Screen Shot 2021-05-13 at 6 23 34 PM

**after**
Screen Shot 2021-05-13 at 6 30 01 PM

### CODE REVIEW

File: game/src/sandbox/dashboards/travel_times.rs [hunk 1]
```diff
@@ -285,51 +286,77 @@ fn contingency_table(ctx: &mut EventCtx, app: &App, filter: &Filter) -> Widget {
...
-        .0
-        .max(savings_per_bucket.iter().max().unwrap().0);
+        .unwrap();
 
     // Draw the bars!
     let bar_width = total_width / (num_buckets as f64);
-    let max_bar_height = (total_height - ctx.default_line_height()) / 2.0;
+    let padded_text_height = ctx.default_line_height() + 12.0;
+    let max_bar_height = (total_height - padded_text_height) / 2.0;
+    let min_bar_height = 8.0;
```
Comment (michaelkirk) [inline game/src/sandbox/dashboards/travel_times.rs:337]: added a min bar height. It was impossible to hover the very small bars.

File: widgetry/src/widgets/compare_times.rs [hunk 2]
```diff
@@ -100,9 +100,11 @@ impl CompareTimes {
                 .collect(),
         )
         .evenly_spaced();
-        let y_label = Text::from(format!("{} (minutes)", y_name.as_ref()))
+        let mut y_label = Text::from(format!("{} (minutes)", y_name.as_ref()))
             .render(ctx)
-            .rotate(Angle::degrees(90.0))
+            .rotate(Angle::degrees(90.0));
+        y_label.autocrop_dims = true;
```
Comment (michaelkirk) [inline widgetry/src/widgets/compare_times.rs:106]: I understand from a code standpoint why this fixes the problem, but I'm not really sure why the "autocrop" method doesn't do this by default.

File: game/src/sandbox/dashboards/mod.rs [hunk 3]
```diff
@@ -34,7 +34,7 @@ impl DashTab {
     pub fn picker(self, ctx: &EventCtx, app: &App) -> Widget {
         let mut choices = vec![
             Choice::new("Trip Table", DashTab::TripTable),
-            Choice::new("Trip Summaries", DashTab::TripSummaries),
+            Choice::new("Travel Times", DashTab::TripDurations),
```
Comment (michaelkirk) [inline game/src/sandbox/dashboards/mod.rs:37]: WDYT of this tab name?

File: game/src/sandbox/dashboards/mod.rs [hunk 4]
```diff
@@ -34,7 +34,7 @@ impl DashTab {
     pub fn picker(self, ctx: &EventCtx, app: &App) -> Widget {
         let mut choices = vec![
             Choice::new("Trip Table", DashTab::TripTable),
-            Choice::new("Trip Summaries", DashTab::TripSummaries),
+            Choice::new("Travel Times", DashTab::TravelTimes),
```
Comment (michaelkirk) [inline game/src/sandbox/dashboards/mod.rs:37]: WDYT of this Tab name?

**Comment by michaelkirk** [comment]
I'm considering adding a toggle to switch the scatter plot to log scale...

File: widgetry/src/widgets/compare_times.rs (hunk 2 above)
Comment (dabreegster) [reply widgetry/src/widgets/compare_times.rs:106]: I had to dig around to remember why we don't always use `autocrop_dims = true`. The problem is with text -- we want to retain the blank space on the left and right sides when laying out text "normally" / not rotated. I know sometimes we add a `Color::CLEAR` rectangle behind things to "force" a proper boundary. If we do this for text, maybe we can get rid of `autocrop_dims` entirely (and act as if it's always true), so then the rotation should just work as expected...

File: game/src/sandbox/dashboards/mod.rs (hunk 4 above)
Comment (dabreegster) [reply game/src/sandbox/dashboards/mod.rs:37]: SGTM

**Comment by dabreegster** [comment]
> I'm considering adding a toggle to switch the scatter plot to log scale...

That'd be a nice feature! The long tail of a few trips lasting over 2 hours kind of hides the interesting patterns in most trips.

Screenshot from 2021-05-14 08-59-51
I was a bit disoriented by the labels not lining up with the bars. I know they mark the boundaries between bars, but with the longer text, it just felt off. I tried extending the tick marks to see if it changes anything, but it doesn't seem to do much:
Screenshot from 2021-05-14 08-59-51
So I think it's all fine, just a thought.

File: game/src/sandbox/dashboards/travel_times.rs [hunk 5]
```diff
@@ -285,51 +286,77 @@ fn contingency_table(ctx: &mut EventCtx, app: &App, filter: &Filter) -> Widget {
...
     let mut outlines = Vec::new();
     let mut tooltips = Vec::new();
     let mut x1 = 0.0;
-    let mut idx = 0;
-    for ((total_savings, num_savings), (total_loss, num_loss)) in savings_per_bucket
+    for (
+        idx,
+        (
+            Changes {
+                accumulated_duration: total_savings,
```
Comment (dabreegster) [inline game/src/sandbox/dashboards/travel_times.rs:345]: I never realized you can do this kind of destructuring in a for loop; wow!

**Comment by michaelkirk** [comment]
> I was a bit disoriented by the labels not lining up with the bars.

Yeah, I see what you mean. Now that units are included in the label it's harder to perceive it as centered on the bar boundaries...

I'll try adding some little lines like in your mockup, I think it could help a little.

**Comment by michaelkirk** [comment]
>> I was a bit disoriented by the labels not lining up with the bars.
>
>Yeah, I see what you mean. Now that units are included in the label, it's harder to perceive it as centered on the bar boundaries...
>
> I'll try adding some little lines like in your mockup, I think it could help a little.

Ok, here's where I landed:
Screen Shot 2021-05-14 at 11 21 28 AM
Screen Shot 2021-05-14 at 11 21 10 AM

WDYT?

I also rebased and included some clippy-fixes for some recently merged code.

**Review by dabreegster — APPROVED** [review]
LGTM. The tick marks help lots!

**Pull request merged**